BitbucketServerAutomation/Functions/Set-BBServerDefaultReviewer.ps1

```diff
@@ -0,0 +1,200 @@
+        # The type of matching to do against the source branch.
+        $SourceBranchType,
+
+        [string]
+        # Specifies the value or pattern to match a source branch, if `-SourceBranchType` is not "Any".
+        $SourceBranchValue,
+
+        [ValidateSet('Any', 'Name', 'Pattern', 'Model')]
+        # The type of matching to do against the target branch.
+        $TargetBranchType,
+
+        [string]
+        # Specifies the value or pattern to match a target branch, if `-TargetBranchType` is not "Any".
+        $TargetBranchValue,
+
+        [object[]]
+        # Collection of objects representing the users to add to the default reviewer condition. Use `Get-BBServerUser` to get Bitbucket Server user objects.
+        $User
+    )
+
+    Set-StrictMode -Version 'Latest'
+    Use-CallerPreference -Cmdlet $PSCmdlet -SessionState $ExecutionContext.SessionState
+
+    if ($SourceBranchValue -and -not $SourceBranchType)
+    {
+        Write-Error -Message 'You must specify a "SourceBranchType" when giving a "SourceBranchValue".'
+        return
+    }
+
+    if ($TargetBranchValue -and -not $TargetBranchType)
+    {
+        Write-Error -Message 'You must specify a "TargetBranchType" when giving a "TargetBranchValue".'
+        return
+    }
+
+    $userProperties = @('name', 'emailAddress', 'id', 'displayName', 'active', 'slug', 'type')
+    foreach ($userObj in $User)
+    {
+        foreach ($property in $userProperties)
+        {
+            if (-not ($userObj | Get-Member -Name $property))
+            {
+                Write-Error -Message '"User" parameter''s argument does not have the expected properties. Use "Get-BBServerUser" to get user objects to pass to this cmdlet.' -ErrorAction $ErrorActionPreference
+                return
+            }
+        }
+    }
+
+    $notFoundErrorMessageSuffix = 'project "{0}"' -f $ProjectKey
+    $repositoryParam = @{ }
+    if ($RepositoryName)
+    {
+        $notFoundErrorMessageSuffix = 'repository "{0}" in project "{1}"' -f $RepositoryName, $ProjectKey
+        $repositoryParam['RepositoryName'] = $RepositoryName
+    }
+
+    $existingCondition =
+        Get-BBServerDefaultReviewer -Connection $Connection -ProjectKey $ProjectKey @repositoryParam |
+        Where-Object { $_ } |
+        Where-Object { $_.id -eq $ID }
+
+    if (-not $existingCondition)
+    {
+        Write-Error -Message ('Default reviewer pull request condition with ID "{0}" does not exist for {1}.' -f $ID, $notFoundErrorMessageSuffix) -ErrorAction $ErrorActionPreference
+        return
+    }
+
+    $requestBody = [PSCustomObject]@{
+        reviewers = @($existingCondition.reviewers | Select-Object -Property $userProperties)
+        sourceMatcher = $existingCondition.sourceRefMatcher
+        targetMatcher = $existingCondition.targetRefMatcher
+        requiredApprovals = $existingCondition.requiredApprovals
+    }
+
+    if ($PSCmdlet.MyInvocation.BoundParameters.ContainsKey('ApprovalCount'))

```

Comment:
Can you replace `$PSCmdlet.MyInvocation.BoundParameters.ContainsKey('ApprovalCount')` with `$PSBoundParameters.ContainsKey('ApprovalCount')`?